crates/kitsune_p2p/kitsune_p2p/src/gossip/sharded_gossip/initiate.rs

```diff
@@ -163,56 +161,74 @@ impl ShardedGossipLocal {
             }
         }
 
-        let windows = self.calculate_time_ranges();
-        let len = windows.len();
-        // Generate the ops bloom for all local agents within the common arc.
-        for (i, window) in windows.into_iter().enumerate() {
-            let blooms = self
-                .generate_ops_blooms_for_time_window(&state.common_arc_set, window)
-                .await?;
-
-            // If no blooms were found for this time window then return a no overlap.
-            if blooms.is_empty() {
-                // Check if this is the final time window.
-                if i == len - 1 {
-                    gossip.push(ShardedGossipWire::ops(
-                        EncodedTimedBloomFilter::NoOverlap,
-                        true,
-                    ));
-                } else {
-                    gossip.push(ShardedGossipWire::ops(
-                        EncodedTimedBloomFilter::NoOverlap,
-                        false,
-                    ));
-                }
+        self.next_bloom_batch(state, gossip).await
+    }
+
+    /// Generate the next batch of blooms from this state.
+    /// If there is a saved cursor from a previous partial
+    /// batch then this will pick up from their.
+    /// Otherwise blooms a bloom for the entire search window
```

Comment:
```suggestion
    /// Otherwise a bloom for the entire search window
```

Reply:
not sure which interpretation is correct :)

```suggestion
    /// Otherwise generates blooms for the entire search window
```

Reply:
```suggestion
    /// Otherwise s batch of blooms for the entire search window
```